components/locale_core/src/extensions/unicode/value.rs

```diff
@@ -61,16 +62,156 @@ impl Value {
         Ok(Self(v))
     }
 
-    #[doc(hidden)]
+    /// Returns a reference to a single [`Subtag`] if the [`Value`] contains exactly one
+    /// subtag, or `None` otherwise.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu::locale::extensions::unicode::Value;
+    /// use std::str::FromStr;
+    ///
+    /// let value1 = Value::from_str("foo")
+    ///     .expect("failed to parse a Value");
+    /// let value2 = Value::from_str("foo-bar")
+    ///     .expect("failed to parse a Value");
+    ///
+    /// assert!(value1.as_single_subtag().is_some());
+    /// assert!(value2.as_single_subtag().is_none());
+    /// ```
     pub const fn as_single_subtag(&self) -> Option<&Subtag> {
         self.0.single()
     }
 
+    /// Destructs into a single [`Subtag`] if the [`Value`] contains exactly one
+    /// subtag, or returns `None` otherwise.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu::locale::extensions::unicode::Value;
+    /// use std::str::FromStr;
+    ///
+    /// let value1 = Value::from_str("foo")
+    ///     .expect("failed to parse a Value");
+    /// let value2 = Value::from_str("foo-bar")
+    ///     .expect("failed to parse a Value");
+    ///
+    /// assert!(value1.into_single_subtag().is_some());
+    /// assert!(value2.into_single_subtag().is_none());
+    /// ```
+    pub fn into_single_subtag(self) -> Option<Subtag> {
+        self.0.into_single()
+    }
+
     #[doc(hidden)]
     pub fn as_subtags_slice(&self) -> &[Subtag] {
         &self.0
     }
 
+    /// Appends a subtag to the back of a [`Value`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu::locale::{
+    ///     extensions::unicode::Value,
+    ///     subtags::subtag,
+    /// };
+    ///
+    /// let mut v = Value::default();
+    /// v.push_subtag(subtag!("foo"));
+    /// v.push_subtag(subtag!("bar"));
+    /// assert_eq!(v, "foo-bar");
+    /// ```
+    pub fn push_subtag(&mut self, subtag: Subtag) {
+        self.0.push(subtag);
+    }
+
+    /// Returns the number of subtags in the [`Value`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu::locale::{
+    ///     extensions::unicode::Value,
+    ///     subtags::subtag,
+    /// };
+    ///
+    /// let mut v = Value::default();
+    /// assert_eq!(v.len(), 0);
+    /// v.push_subtag(subtag!("foo"));
+    /// assert_eq!(v.len(), 1);
+    /// ```
+    pub fn len(&self) -> usize {
```

Comment:
Nit: There is some ambiguity of what the `len` is (could be # of characters) so I usually prefer more specific function names such as `subtag_count()`